src/other/precompile.jl

```diff
@@ -1,6 +1,14 @@
 import SnoopPrecompile
 
 SnoopPrecompile.@precompile_all_calls begin
+    import CSV
```

Comment:
@timholy

The question is when precompilation is enabled.
Could you explain why CSV.jl is visible as a module when I e.g. write `DataFrames.CSV` in consecutive loads of DataFrames.jl?

I thought that code inside `SnoopPrecompile.@precompile_all_calls` runs only when the package is being precompiled. However, it seems that somehow the fact that I do `import CSV` is cached and later is visible (even in consecutive sessions).

The same is with the definition of extra `Base.reduce` method.

Thank you!

Reply:
Right. If your package forces loading of CSV at precompile time, then there might be dispatches that link against CSV. So you can't suddenly rip CSV out and expect all the code to be happy. You'd basically end up with a corrupt, unresolvable pkgimage.